src/Data/Blob.hs

```diff
@@ -44,12 +45,12 @@ data File = File
   , fileLanguage :: Language
   } deriving (Show, Eq, Generic)
 
--- | Prefer 'fileForRelPath' if at all possible.
+-- | Prefer 'fileForTypedPath' if at all possible.
 fileForPath :: FilePath  -> File
 fileForPath p = File p (languageForFilePath p)
 
-fileForRelPath :: Path.RelFile -> File
-fileForRelPath = fileForPath . Path.toString
+fileForTypedPath :: Path.PartClass.AbsRel ar => Path.File ar -> File
+fileForTypedPath = fileForPath . Path.toString
```

Comment:
This now admits a more general type—any of `Path.AbsFile`, `Path.RelFile`, or `Path.AbsRelFile` will suffice. As such, I changed its name.